### ISSUE TITLE
Clang-Format inserts extra indent for initializer lists with PP in the #else branch

### ISSUE THREAD

**Issue by petrhosek** [llvm/llvm-project#69237]
For the following snippet:
```cpp
A a_ {
  kFlag1,
#if BUILD_FLAG
      kFlag2,
#else
      kFlag3,
#endif
      kFlag4
};
```
Clang-Format inserts an extra indent in the `#else` branch after #65409:
```cpp
A a_{kFlag1,
#if BUILD_FLAG
     kFlag2,
#else
       kFlag3,
#endif
     kFlag4};
```

**Comment by owenca** [comment]
@rymiel

**Comment by owenca** [comment]
May be a duplicate of #68404.

**Comment by rymiel** [comment]
It's unrelated, I haven't yet found the cause for this one yet
